src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -597,17 +597,39 @@ public static async ValueTask<FormulaValue> AverageTable(EvalVisitor runner, Eva
         }
 
         // https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-mod
-        public static FormulaValue Mod(IRContext irContext, NumberValue[] args)
+        public static FormulaValue Mod(IRContext irContext, FormulaValue[] args)
         {
-            var arg0 = args[0].Value;
-            var arg1 = args[1].Value;
-
+            double arg0 = ((NumberValue)args[0]).Value;
+
+            if (args[1] is TableValue table)
+            {
+                List<FormulaValue> modResults = new List<FormulaValue>();
+
+                foreach (DValue<RecordValue> row in table.Rows)
+                {
+                    FormulaValue val = row.Value.GetField("Value");
+                    double arg1 = val is BlankValue ? 0d : ((NumberValue)val).Value;

```

Comment:
consider inverting, could be cleaner?
`val is NumberValue numVal ? numVal.Value : 0d`

Reply:
fixed